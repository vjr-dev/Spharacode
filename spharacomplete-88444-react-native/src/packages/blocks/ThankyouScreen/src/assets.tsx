export const back1 = require("../assets/image_background-1.png");
export const back2 = require("../assets/image_background-2.png");

export const LOGO = require("../assets/LOGO.png");
export const Thanks = require("../assets/thanks.png");
export const Rectangle = require("../assets/view_Rectangle.png");


