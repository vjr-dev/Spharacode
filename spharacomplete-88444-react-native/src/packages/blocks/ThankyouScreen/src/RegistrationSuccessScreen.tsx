// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";
import {
  StyleSheet,
  Text,
  View,
  Image,
  ImageBackground,
  SafeAreaView,
  TouchableOpacity,
} from "react-native";
import RegistrationSuccessScreenController, {
  Props,
} from "./RegistrationSuccessScreenController";
import { back1, back2, Thanks } from "./assets";
import { COLORS } from "framework/src/Globals";
import { deviceHeight, deviceWidth, scaledSize } from "framework/src/Utilities";
import Loader from "../../../components/src/Loader";

export default class RegistrationSuccessScreen extends RegistrationSuccessScreenController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <SafeAreaView style={styles.container}>
        <ImageBackground source={back1} style={styles.image}>
          <ImageBackground source={back2} style={styles.image2}>
            <View>
              <View style={styles.mainView}>
                <View style={styles.subView}>
                  <Text style={[styles.text1, { marginTop: scaledSize(150) }]}>
                    Registration Successfully finished{" "}
                  </Text>
                  <Image
                    source={Thanks}
                    resizeMode="contain"
                    style={styles.logo}
                  />
                  <Text style={styles.whatNextText}>What is next?</Text>
                  <Text style={[styles.text2, { fontSize: scaledSize(12) }]}>
                    In order to start receiving emergency calls you need to
                    complete your profile and upload your first responder
                    certificate.
                  </Text>
                </View>
              </View>

              <View style={styles.btnView}>
                <TouchableOpacity
                  testID="onCompleteProfileButton"
                  onPress={() => this.onCompleteProfile()}
                  style={styles.btn}
                >
                  <Text style={styles.btnText}>COMPLETE PROFILE</Text>
                </TouchableOpacity>
              </View>
            </View>
            <Loader loading={this.state.isLoading} />
          </ImageBackground>
        </ImageBackground>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#454545",
  },
  image: {
    height: "100%",
    width: "100%",
  },
  image2: {
    height: "100%",
    width: "100%",
    alignItems: "center",
  },
  mainView: {
    alignItems: "center",
    height: "100%",
    flex: 1,
  },
  subView: {
    flex: 1,
    padding: scaledSize(10),
    alignItems: "center",
  },
  logo: {
    height: scaledSize(100),
    width: scaledSize(100),
    marginTop: scaledSize(30),
  },
  text1: {
    fontSize: scaledSize(18),
    textAlign: "center",
    color: COLORS.white,
    marginTop: scaledSize(25),
    fontWeight: "bold",
    letterSpacing: 1,
    width: deviceWidth - scaledSize(90),
  },
  whatNextText: {
    fontSize: scaledSize(16),
    textAlign: "center",
    color: COLORS.white,
    marginTop: scaledSize(25),
    width: deviceWidth - scaledSize(90),
  },
  btnText: {
    fontSize: scaledSize(14),
    color: COLORS.white,
    fontWeight: "bold",
    letterSpacing: 1,
    textAlign: "center",
  },
  text2: {
    fontSize: scaledSize(13),
    color: COLORS.infoGray,
    marginTop: 10,
    width: deviceWidth - scaledSize(90),
    fontWeight: "600",
    textAlign: "center",
    alignSelf: "center",
  },
  btnView: {
    alignItems: "center",
    width: deviceWidth,
    justifyContent: "flex-end",
    marginBottom: deviceHeight * 0.05,
  },
  btn: {
    height: scaledSize(50),
    width: deviceWidth - scaledSize(20),
    backgroundColor: COLORS.darkorange,
    borderRadius: scaledSize(30),
    justifyContent: "center",
  },
});

// Customizable Area End
