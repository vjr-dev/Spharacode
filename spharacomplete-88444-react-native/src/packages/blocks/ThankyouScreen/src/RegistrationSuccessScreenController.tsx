// Customizable Area Start
//@ts-ignore
//@ts-nocheck

import { IBlock } from "framework/src/IBlock";
import { BlockComponent } from "framework/src/BlockComponent";
import { runEngine } from "framework/src/RunEngine";
import { BackHandler } from "react-native";
import { backToLoginConfirmationAlert } from "../../../components/src/CustomAlert";
import { OnLogOut } from "../../../components/src/Navigation/logout";
import AsyncStorage from "@react-native-async-storage/async-storage";

export interface Props {
  navigation: any;
  id: string;
}

interface S {
  isLoading: boolean;
}

interface SS {
  id: any;
}

export default class RegistrationSuccessScreenController extends BlockComponent<
  Props,
  S,
  SS
> {
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    this.subScribedMessages = [];

    this.state = {
      isLoading: false,
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async componentDidMount(): Promise<void> {
    const tempRoleID = await AsyncStorage.getItem("roleID");
    this.setState({ roleID: tempRoleID });
  }

  componentWillMount() {
    BackHandler.addEventListener(
      "hardwareBackPress",
      this.handleBackButtonClick
    );
  }

  componentWillUnmount() {
    BackHandler.removeEventListener(
      "hardwareBackPress",
      this.handleBackButtonClick
    );
  }

  handleBackButtonClick = async () => {
    backToLoginConfirmationAlert(async () => {
      this.setState({ isLoading: true });
      await OnLogOut(this.props);
      this.setState({ isLoading: false });
    });
    return true;
  };

  onCompleteProfile() {
    this.props.navigation.replace("PermissionScreen");
  }
}
// Customizable Area End
