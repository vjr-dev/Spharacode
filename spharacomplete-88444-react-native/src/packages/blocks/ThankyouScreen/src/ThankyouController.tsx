// Customizable Area Start
//@ts-ignore
//@ts-nocheck

import { IBlock } from "../../../framework/src/IBlock";
import { BlockComponent } from "../../../framework/src/BlockComponent";

import { runEngine } from "../../../framework/src/RunEngine";
import { BackHandler } from "react-native";
import { backToLoginConfirmationAlert } from "../../../components/src/CustomAlert";
import { OnLogOut } from "../../../components/src/Navigation/logout";
import AsyncStorage from "@react-native-async-storage/async-storage";

export interface Props {
  navigation: any;
  id: string;
}

interface S {
  data: any;
  roleID: string;
  istooltipVisible: boolean;
  isLoading: boolean;
}

interface SS {
  id: any;
}

export default class TutorialsController extends BlockComponent<Props, S, SS> {
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    this.subScribedMessages = [];

    this.state = {
      data: [1, 2, 3, 4, 5],
      roleID: "",
      istooltipVisible: false,
      isLoading: false,
    };
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }

  async componentDidMount(): Promise<void> {
    const tempRoleID = await AsyncStorage.getItem("roleID");
    this.setState({ roleID: tempRoleID });
  }

  componentWillMount() {
    BackHandler.addEventListener(
      "hardwareBackPress",
      this.handleBackButtonClick
    );
  }

  componentWillUnmount() {
    BackHandler.removeEventListener(
      "hardwareBackPress",
      this.handleBackButtonClick
    );
  }

  handleBackButtonClick = async () => {
    backToLoginConfirmationAlert(async () => {
      this.setState({ isLoading: true });
      await OnLogOut(this.props);
      this.setState({ isLoading: false });
    });
    return true;
  };

  onclick() {
    const { navigation }: any = this.props;
    if (this.state.roleID === "1") {
      navigation.replace("Authentication");
    } else {
      navigation.replace("PermissionScreen");
    }
  }
  goback() {
    const { navigation }: any = this.props;
    navigation.goBack();
  }
}
// Customizable Area End
