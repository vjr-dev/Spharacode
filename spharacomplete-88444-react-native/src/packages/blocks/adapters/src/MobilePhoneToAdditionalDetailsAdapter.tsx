import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
