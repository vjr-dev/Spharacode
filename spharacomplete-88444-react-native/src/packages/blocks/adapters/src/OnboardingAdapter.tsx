import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class OnboardingAdapter {
  send: (message: Message) => void;

  constructor() {
    // const uuidv4 = require('uuid/v4');
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.OnboardingGuideControllerDoneOnboarding),
      getName(MessageEnum.OnboardingGuideControllerSkipOnboarding)
    ]);
  }

  convert = (from: Message): Message => {
    runEngine.debugLog("OnboardingAdapter", from.id)
    const to = new Message(getName(MessageEnum.NavigationMessage));
    to.addData(getName(MessageEnum.NavigationTargetMessage), 'Home');
    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );
    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
