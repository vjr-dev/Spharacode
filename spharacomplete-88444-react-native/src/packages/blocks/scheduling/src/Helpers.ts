/* App/Lib/GeneralHelpers.js */
import { Platform, StyleSheet } from 'react-native';
import { heightFromPercentage } from '../../../framework/src/Utilities';

// Colors Constants
export const COLORS = {
  TRANSPARENT: 'transparent',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  black10: 'rgb(10,10,10)',
  green: 'rgb(76,217,100)',
  red: 'red',
  yellow: 'rgba(255, 212, 40, 1)',
  darkGray: 'rgba(105,105,105,0.9)',
  lightGray: '#a7b6b9',
  skipGray: 'rgb(91,91,92)',
  borderColor: 'rgb(240, 240, 240)',
  inactiveIndicator: 'rgb(228, 233, 242)',
  inputLabel: 'rgb(143,155,179)',
  inputIcon: 'rgb(197,206,224)',
  inputValue: 'rgb(46,58,89)',
  discountRed: 'rgb(199,5,28)',
  orange: 'rgb(255,157,43)',
  infoGray: 'rgb(163,164,165)',
  ReviewBlack: 'rgb(34,43,69)',
  blue: 'blue',
  blueGrey: 'rgb(143,155,179)',
  darkBlueGrey: 'rgb(46,58,89)',
  lightBlueGrey: 'rgb(197, 206, 224)'
};

export const CommonStyle = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  wrapper: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  headerContainer:{
    backgroundColor:COLORS.yellow,
    height: '10%',
  },
  authHeader:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.yellow
  },
  authHeaderWhite:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.white
  },
  formContainer: {
    width: "80%"
  },
  textAlignCenter:{
    textAlign: 'center'
  },
  labelTextStyle: {
    color: COLORS.inputLabel
  },
  inputContainerStyle:{
    borderBottomWidth: 1,
    borderBottomColor: COLORS.inactiveIndicator
  },
  error: {
    color: COLORS.red,
    fontSize: 14
  },
  scrollStyle: {
    width: "100%"
  },
  btnTitleStyle:{
    fontSize: 17,
    color: COLORS.black
  },
  closeButtonStyle: {
    backgroundColor: COLORS.inputIcon,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  yellowButtonStyle: {
    backgroundColor: COLORS.yellow,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  closeBtnTitleStyle:{
    color: COLORS.black
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: "center"
  },
  shadowStyle: {
    shadowOffset:
      Platform.OS === "ios"
        ? { width: 0, height: 1 }
        : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 3 : 5,
    elevation: 5
  },
  boxShadowStyle: {
    backgroundColor: COLORS.white,
    marginHorizontal:2,
    marginVertical:1,
    shadowOffset:
    Platform.OS === "ios"
      ? { width: 0, height: 1 }
      : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 1 : 5,
    elevation: 2,    
    borderRadius:8,
  },
  toastModal: {
    borderRadius: 20,
    marginHorizontal: 15,
    paddingHorizontal: '12%',
    paddingVertical: 100,
    alignContent: "center",
    justifyContent: "center",
  },
  clearModal: {
    borderRadius: 5,
    marginHorizontal: 50,
    height: 220,
    backgroundColor: COLORS.red
  },
  imageThumbContain: {
    resizeMode: "contain",
    width: "100%",
    height: "100%"
  },
  imageThumbCover: {
    resizeMode: "cover",
    width: "100%",
    height: "100%"
  },
  flexContainer: {
    flex: 1,
    width: "100%"
  },
  rowStyle: {
    flexDirection: "row"
  },
  flexStart: {
    justifyContent: "flex-start"
  },
  flexEnd: {
    justifyContent: "flex-end"
  },
  jCCenter: {
    justifyContent: "center"
  },
  spaceBetween: {
    justifyContent: "space-between",    
  },
  spaceAround: {
    justifyContent: 'space-around'
  },
  alignItemsCenter:{
    alignItems: "center"
  },
  styleCenter: {
    justifyContent: "flex-end",
    alignItems: "center"
  },
  centerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  width100P:{
    width: '100%'
  },
  subView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  emptyList: {
    alignItems: "center",
    marginTop: 100
  },
  emptyListFilter: {
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100
  },
  backIconView: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: 30,
    height: 30,
    justifyContent: "flex-end",
    alignItems: "flex-end"
  },
  crossImage: {
    tintColor: COLORS.red,
    width: "100%",
    height: "100%",
    resizeMode: "contain"
  },
  searchTextInput: {
    width: "80%"
  },
  errorReport: {
    width: "70%",
    color: COLORS.red,
    fontSize: 15,
    marginLeft: "28.8%"
  }
});