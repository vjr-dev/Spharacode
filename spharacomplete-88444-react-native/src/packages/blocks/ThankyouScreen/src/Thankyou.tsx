// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import React from "react";

import {
  StyleSheet,
  Text,
  View,
  Image,
  ImageBackground,
  SafeAreaView,
  TouchableOpacity,
} from "react-native";

import ThankyouController, { Props } from "./ThankyouController";
import { back1, back2, Thanks } from "./assets";
import { COLORS } from "framework/src/Globals";
import { deviceHeight, deviceWidth, scaledSize } from "framework/src/Utilities";
import Loader from "../../../components/src/Loader";
export default class Thankyou extends ThankyouController {
  constructor(props: Props) {
    super(props);
  }

  render() {
    return (
      <SafeAreaView style={styles.container}>
        <ImageBackground source={back1} style={styles.image}>
          <ImageBackground source={back2} style={styles.image2}>
            <TouchableOpacity
              testID={"outsideClick"}
              onPress={() => this.setState({ istooltipVisible: false })}
              activeOpacity={1}
            >
              <View style={styles.mainView}>
                <View style={styles.subView}>
                  <Image
                    source={Thanks}
                    resizeMode="contain"
                    style={styles.logo}
                  />
                  <Text style={styles.text1}>Thank You!</Text>
                  <Text style={styles.text2}>
                    Your phone number has been verified successfully.
                  </Text>
                </View>

                {this.state.roleID == "1" && (
                  <View style={styles.authCodeView}>
                    {this.state.istooltipVisible ? (
                      <View style={styles.tooltipContainer}>
                        <Text testID={"popupText"} style={styles.tooltipText}>
                          Authentication code is a 6 digit code sent by us to
                          your company which confirm your identity as first
                          respoder.
                        </Text>
                        <View style={styles.triangle} />
                      </View>
                    ) : null}
                    <Text style={styles.authText}>
                      {"Next, please enter "}
                      <Text
                        testID="authCodeLink"
                        onPress={() =>
                          this.setState({ istooltipVisible: true })
                        }
                        style={styles.orangeText}
                      >
                        {"authentication code "}
                      </Text>
                      {"sent \nyou as SMS by your organization"}
                    </Text>
                  </View>
                )}
              </View>

              <View style={styles.btnView}>
                <TouchableOpacity
                  testID="continueButton"
                  onPress={() => this.onclick()}
                  style={styles.btn}
                >
                  <Text style={styles.btnText}>
                    {" "}
                    {this.state.roleID == "1"
                      ? "CONTINUE"
                      : "CONTINUE TO PROFILE"}
                  </Text>
                </TouchableOpacity>
              </View>
            </TouchableOpacity>
            <Loader loading={this.state.isLoading} />
          </ImageBackground>
        </ImageBackground>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#454545",
  },
  image: {
    height: "100%",
    width: "100%",
  },
  image2: {
    height: "100%",
    width: "100%",
    alignItems: "center",
  },
  mainView: {
    height: "100%",
    flex: 1,
    justifyContent: "center",
  },
  subView: {
    padding: scaledSize(10),
    alignItems: "center",
    marginTop: scaledSize(50),
  },
  logo: {
    height: scaledSize(100),
    width: scaledSize(100),
  },
  text1: {
    fontSize: scaledSize(18),
    textAlign: "center",
    color: COLORS.white,
    marginTop: scaledSize(25),
    fontWeight: "bold",
    letterSpacing: 1,
  },
  text2: {
    fontSize: scaledSize(13),
    color: COLORS.infoGray,
    marginTop: 10,
    width: deviceWidth - scaledSize(90),
    fontWeight: "600",
    textAlign: "center",
    alignSelf: "center",
  },
  authText: {
    fontSize: scaledSize(12),
    color: COLORS.infoGray,
    width: deviceWidth - scaledSize(90),
    fontWeight: "600",
    textAlign: "center",
    alignSelf: "center",
    lineHeight: scaledSize(20),
  },
  orangeText: {
    color: COLORS.orangelight,
  },
  btnView: {
    alignItems: "center",
    width: deviceWidth,
    justifyContent: "flex-end",
    marginBottom: deviceHeight * 0.05,
  },
  btn: {
    height: scaledSize(50),
    width: deviceWidth - scaledSize(20),
    backgroundColor: COLORS.darkorange,
    borderRadius: scaledSize(30),
    justifyContent: "center",
  },
  btnText: {
    fontSize: scaledSize(14),
    color: COLORS.white,
    fontWeight: "bold",
    letterSpacing: 1,
    textAlign: "center",
  },
  authCodeView: {
    marginTop: scaledSize(100),
    width: "auto",
    alignSelf: "center",
  },
  tooltipContainer: {
    position: "absolute",
    right: scaledSize(10),
    top: scaledSize(-75),
    width: scaledSize(170),
    padding: scaledSize(5),
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: COLORS.orangelight,
    borderRadius: 5,
  },
  triangle: {
    position: "absolute",
    bottom: scaledSize(-10),
    width: 0,
    height: 0,
    backgroundColor: "transparent",
    borderStyle: "solid",
    borderLeftWidth: scaledSize(5),
    borderRightWidth: scaledSize(5),
    borderBottomWidth: scaledSize(10),
    borderLeftColor: "transparent",
    borderRightColor: "transparent",
    borderBottomColor: COLORS.orangelight,
    transform: [{ rotate: "180deg" }],
  },
  tooltipText: {
    fontSize: scaledSize(10),
    color: COLORS.black,
  },
});

// Customizable Area End
