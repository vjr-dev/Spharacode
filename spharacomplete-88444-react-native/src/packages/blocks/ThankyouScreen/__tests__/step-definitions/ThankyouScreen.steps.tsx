//@ts-ignore
//@ts-nocheck
import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "framework/src/Helpers";
import React from "react";
import Thankyou from "../../src/Thankyou";
import { BackHandler } from "react-native";
import AsyncStorage from "@react-native-async-storage/async-storage";

const screenProps = {
  navigation: {
    navigate: jest.fn(),
    replace: jest.fn(),
    goBack: jest.fn(),
    addListener: jest.fn().mockImplementation((event: any, callback: any) => {
      callback();
    }),
  },
  route: {
    params: {
      data: { name: "Test Name" },
      userName: "abc",
    },
  },
  id: "ThankyouScreen",
};

const feature = loadFeature(
  "./__tests__/features/ThankyouScreen-scenario.feature"
);
jest.mock("react-native/Libraries/Utilities/BackHandler", () => {
  return jest.requireActual(
    "react-native/Libraries/Utilities/__mocks__/BackHandler.js"
  );
});
defineFeature(feature, (test) => {
  const mockAsyncStorage = jest.spyOn(AsyncStorage, "getItem");
  beforeEach(() => {
    jest.resetModules();
    jest.doMock("react-native", () => ({
      Platform: { OS: "web" },
      BackHandler: {
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
      },
    }));
    jest.spyOn(helpers, "getOS").mockImplementation(() => "web");
  });
  test("User navigates to ThankyouScreen", ({ given, when, then }) => {
    let ThankyouScreenBlockWrapper: ShallowWrapper;
    let instance: Thankyou;

    given("I am a User loading ThankyouScreen", () => {
      //@ts-ignore
      ThankyouScreenBlockWrapper = shallow(<Thankyou {...screenProps} />);
      expect(ThankyouScreenBlockWrapper).toBeTruthy();
      expect(ThankyouScreenBlockWrapper).toMatchSnapshot();
    });

    when("I navigate to the ThankyouScreen", () => {
      instance = ThankyouScreenBlockWrapper.instance() as Thankyou;
    });

    then("ThankyouScreen will load with out errors", () => {
      //@ts-ignore
      expect(ThankyouScreenBlockWrapper).toBeTruthy();
    });

    then("Handle onClick button press", () => {
      let buttonComponent = ThankyouScreenBlockWrapper.findWhere(
        (node) => node.prop("testID") === "continueButton"
      );
      buttonComponent.simulate("press", instance.onclick());
      instance.goback();
      //@ts-ignore
      expect(buttonComponent).toBeTruthy();
    });
    then(
      "User can check authentication code popup if user is first responder",
      async () => {
        mockAsyncStorage.mockClear();
        mockAsyncStorage.mockImplementation((key) => {
          if (key == "roleID") {
            return "1";
          }
        });
        const wrapper = await shallow(<Thankyou {...screenProps} />);
        let authCodeLink = wrapper.findWhere(
          (node) => node.prop("testID") === "authCodeLink"
        );
        authCodeLink.simulate("press");
        let popupText = wrapper.findWhere(
          (node) => node.prop("testID") === "popupText"
        );
        expect(popupText.props().children).toBe(
          "Authentication code is a 6 digit code sent by us to your company which confirm your identity as first respoder."
        );
      }
    );
    then(
      "User can close authentication code popup if user click outside",
      async () => {
        mockAsyncStorage.mockClear();
        mockAsyncStorage.mockImplementation((key) => {
          if (key == "roleID") {
            return "1";
          }
        });
        const wrapper = await shallow(<Thankyou {...screenProps} />);
        let authCodeLink = wrapper.findWhere(
          (node) => node.prop("testID") === "authCodeLink"
        );
        authCodeLink.simulate("press");
        let outsideClick = wrapper.findWhere(
          (node) => node.prop("testID") === "outsideClick"
        );
        outsideClick.simulate("press");
        let popupText = wrapper.findWhere(
          (node) => node.prop("testID") === "popupText"
        );
        expect(popupText.isEmptyRender()).toBe(true);
      }
    );
    then("I can unmount screen without error", () => {
      BackHandler.mockPressBack();
      ThankyouScreenBlockWrapper.unmount();
    });
  });
});
