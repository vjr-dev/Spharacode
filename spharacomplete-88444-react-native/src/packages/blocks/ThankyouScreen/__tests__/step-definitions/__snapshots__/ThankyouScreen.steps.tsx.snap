// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankyouScreen User navigates to ThankyouScreen 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <ImageBackground
    source={
      Object {
        "testUri": "../../../packages/blocks/ThankyouScreen/assets/image_background-1.png",
      }
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/ThankyouScreen/assets/image_background-2.png",
        }
      }
      style={
        Object {
          "alignItems": "center",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <ForwardRef
        activeOpacity={1}
        onPress={[Function]}
        testID="outsideClick"
      >
        <View
          style={
            Object {
              "flex": 1,
              "height": "100%",
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "marginTop": 96,
                "padding": 20,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../packages/blocks/ThankyouScreen/assets/thanks.png",
                }
              }
              style={
                Object {
                  "height": 191,
                  "width": 191,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 35,
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "marginTop": 48,
                  "textAlign": "center",
                }
              }
            >
              Thank You!
            </Text>
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "rgb(163,164,165)",
                  "fontSize": 25,
                  "fontWeight": "600",
                  "marginTop": 10,
                  "textAlign": "center",
                  "width": 578,
                }
              }
            >
              Your phone number has been verified successfully.
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "flex-end",
              "marginBottom": 66.7,
              "width": 750,
            }
          }
        >
          <ForwardRef
            onPress={[Function]}
            style={
              Object {
                "backgroundColor": "rgba(255,138,73,1)",
                "borderRadius": 58,
                "height": 96,
                "justifyContent": "center",
                "width": 711,
              }
            }
            testID="continueButton"
          >
            <Text
              style={
                Object {
                  "color": "rgb(255,255,255)",
                  "fontSize": 27,
                  "fontWeight": "bold",
                  "letterSpacing": 1,
                  "textAlign": "center",
                }
              }
            >
               
              CONTINUE TO PROFILE
            </Text>
          </ForwardRef>
        </View>
      </ForwardRef>
      <Loader
        loading={false}
      />
    </ImageBackground>
  </ImageBackground>
</ForwardRef(SafeAreaView)>
`;
